//package core.io.protocol;

import "BattleEnemyUnit.proto"; 
import "BattleSelfUnit.proto";
import "BattleOneUnit.proto";

message BattleAction
{
	optional string		  		attackid   		= 1;//本次攻击式神ID
	optional int32		  		attackrole		= 2;//1:攻击方 2:防守方
	optional int32		  		skillid       	= 3;//本次攻击技能ID
	optional int32				skillparam		= 4;//技能参数
	repeated BattleEnemyUnit 	targetlist    	= 5;//攻击目标
	repeated BattleSelfUnit		owneffectlist	= 6;//对己方生效目标
	repeated BattleOneUnit  	allspiritslist	= 7;//双方式神数据同步
}

